### PR TITLE
Fix several panics in ParseCalendar

### PR DESCRIPTION
**Pull request by brackendawson**
I was trying to Fuzz [webcal-proxy](https://github.com/brackendawson/webcal-proxy) but I need golang-ical to be panic free first.

### CODE REVIEW

**Comment by arran4** [comment]
Thanks. Looks good. Running tests.

File: calendar_test.go [hunk 1]
```diff
@@ -360,3 +363,13 @@ func TestIssue52(t *testing.T) {
...
 	}
 }
+
+func FuzzParseCalendar(f *testing.F) {
+	ics, err := os.ReadFile("testdata/timeparsing.ics")
+	require.NoError(f, err)
+	f.Add(ics)
+	f.Fuzz(func(t *testing.T, ics []byte) {
+		_, err := ParseCalendar(bytes.NewReader(ics))
+		t.Log(err)
```
Comment (arran4) [inline calendar_test.go:373]: Shouldn't this be t.Errorf?

**Review by arran4 — APPROVED** [review]

**Comment by arran4** [comment]
Tests are failing for pre-fuzzing go. Perhaps use build tags?

Something like:
```
// +build go1.18
```

But that is not correct as it isn't forwards compatible I think. I rarely use go version build constraints

**Comment by arran4** [comment]
Hrm.. Should modify the github workflow action to test on more recent versions of go as well.

**Comment by brackendawson** [comment]
> Hrm.. Should modify the github workflow action to test on more recent versions of go as well.

I notice you test from 1.14 but the go directive in the mod file is 1.13.

**Comment by arran4** [comment]
Cool all tests pass.

> I notice you test from 1.14 but the go directive in the mod file is 1.13.

Interesting. I didn't notice. That would have happened by accident over time rather than intentional. I don't see a reason to change it right now but will accept a PR/change that pushes the matrix back.

File: calendar_test.go (hunk 1 above)
Comment (arran4) [reply calendar_test.go]: Acknowledged. I haven't used the fuzzing feature myself. Not nearly enough go coding these days.

**Review by arran4 — APPROVED** [review]
LGTM

**Comment by arran4** [comment]
```
% git-tag-inc release && git push --tags
Largest: v0.2.3 (51fa6f12130177fd8e6ef8e6e5fcc79e28141324)
Creating v0.2.4
warning: not sending a push certificate since the receiving end does not support --signed push
Enumerating objects: 1, done.
Counting objects: 100% (1/1), done.
Writing objects: 100% (1/1), 155 bytes | 155.00 KiB/s, done.
Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
To https://github.com/arran4/golang-ical.git
 * [new tag]         v0.2.4 -> v0.2.4
```

Tagged to release for you

**Pull request merged**